src/test/java/bio/terra/service/dataset/DatasetDaoTest.java

```diff
@@ -289,7 +289,7 @@ public void mixingSharedAndExclusiveLocksTest() throws IOException, SQLException
 
             // 1. take out a shared lock
             // confirm that there are no exclusive locks and one shared lock
-            datasetDao.lockShared(datasetId, "flightid1");
+            datasetDao.lockShared(datasetId, "flightid1", null);
```

Comment:
I think lockShared should either have an overloaded method that specifies null as the third argument, or move the configuration service check for the fault into datasetdao per comment above so that you don't need the argument.

Reply:
No longer needed with your other suggestion!